test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -850,12 +850,14 @@ protected void assertFetchMany(MapperService mapperService, String field, Object
      */
     protected void assertFetch(MapperService mapperService, String field, Object value, String format) throws IOException {
         MappedFieldType ft = mapperService.fieldType(field);
-        MappedFieldType.FielddataOperation fdt = MappedFieldType.FielddataOperation.SEARCH;
         SourceToParse source = source(b -> b.field(ft.name(), value));
+        var fielddataContext = useSearchOperationForFetchTests()
```

Comment:
Given that this is just for pattern_text, maybe overwrite this method in `PatternTextFieldMapperTests`?
And create fetcher with field data context for scripts there. That way there is no need to change this base class?